packages/wonder-blocks-core/src/hooks/use-unique-id.js

```diff
@@ -0,0 +1,65 @@
+// @flow
+import * as React from "react";
+
+import SsrIDFactory from "../util/ssr-id-factory.js";
+import UniqueIDFactory from "../util/unique-id-factory.js";
+import {useRenderState} from "./use-render-state.js";
+
+import type {IIdentifierFactory} from "../util/types.js";
+
+/**
+ * Returns a unique identifier factory.  If the parent component hasn't
+ * been mounted yet, the global SsrIDFactory will be returned until the
+ * component becomes mounted.
+ *
+ * @param {string} [scope] optional string to prefix generated ids with.
+ * @returns {IIdentifierFactory}
+ */
+export const useUniqueIdWithMock = (scope?: string): IIdentifierFactory => {
+    const renderState = useRenderState();
+    const [isMounted, setIsMounted] = React.useState(false);
```

Comment:
Huh, do we have to use `React.` for all hooks? Is there a way to avoid that?

Reply:
I can do named imports.  I went with `React.` because we do namespace imports of "react" everywhere in wonder-blocks.